From d10d613b33ab169913a376a6c3e5a641e2ea1189 Mon Sep 17 00:00:00 2001
From: Hannah von Reth <vonreth@kde.org>
Date: Wed, 20 Dec 2017 09:51:41 +0100
Subject: [PATCH] Remove anchient and broken workaround

Summary: The Craft version of aspell is relocatable and this hack is not

Reviewers: #frameworks, #craft, leinir, apol

Reviewed By: leinir, apol

Subscribers: apol, #frameworks

Tags: #frameworks

Differential Revision: https://phabricator.kde.org/D9421
---
 src/plugins/aspell/CMakeLists.txt   |  3 ---
 src/plugins/aspell/aspellclient.cpp | 15 ---------------
 src/plugins/aspell/aspelldict.cpp   |  6 ------
 3 files changed, 24 deletions(-)

diff --git a/src/plugins/aspell/CMakeLists.txt b/src/plugins/aspell/CMakeLists.txt
index 3b6ad96..a29aab4 100644
--- a/src/plugins/aspell/CMakeLists.txt
+++ b/src/plugins/aspell/CMakeLists.txt
@@ -7,9 +7,6 @@ include_directories( ${ASPELL_INCLUDE_DIR})
 set(sonnet_aspell_PART_SRCS aspellclient.cpp aspelldict.cpp)
 ecm_qt_declare_logging_category(sonnet_aspell_PART_SRCS HEADER aspell_debug.h IDENTIFIER SONNET_LOG_ASPELL CATEGORY_NAME sonnet.plugins.aspell)

-if(WIN32)
-add_definitions(-DCMAKE_INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX}")
-endif()
 add_library(sonnet_aspell MODULE ${sonnet_aspell_PART_SRCS})

 target_link_libraries(sonnet_aspell PRIVATE KF5::SonnetCore ${ASPELL_LIBRARIES})
diff --git a/src/plugins/aspell/aspellclient.cpp b/src/plugins/aspell/aspellclient.cpp
index 1949730..9a95fe7 100644
--- a/src/plugins/aspell/aspellclient.cpp
+++ b/src/plugins/aspell/aspellclient.cpp
@@ -27,25 +27,10 @@

 using namespace Sonnet;

-#ifdef Q_OS_WIN
-#define ASPELL_DATA_ROOT "/lib/aspell-0.60/"
-
-QString aspell_data_dir()
-{
-    // TODO: is this still valid on Windows?
-    // A generated config-aspell.h (or config-kspell.h, if shared) should be added then, to define CMAKE_INSTALL_PREFIX
-    return QLatin1String(CMAKE_INSTALL_PREFIX  ASPELL_DATA_ROOT);
-}
-#endif
-
 ASpellClient::ASpellClient(QObject *parent)
     : Client(parent)
 {
     m_config = new_aspell_config();
-#ifdef Q_OS_WIN
-    aspell_config_replace(m_config, "data-dir", aspell_data_dir().toLocal8Bit().data());
-    aspell_config_replace(m_config, "dict-dir", aspell_data_dir().toLocal8Bit().data());
-#endif
 }

 ASpellClient::~ASpellClient()
diff --git a/src/plugins/aspell/aspelldict.cpp b/src/plugins/aspell/aspelldict.cpp
index f008f49..cdff226 100644
--- a/src/plugins/aspell/aspelldict.cpp
+++ b/src/plugins/aspell/aspelldict.cpp
@@ -35,12 +35,6 @@ ASpellDict::ASpellDict(const QString &lang)
     /* For reference, please look at BR#87250         */
     aspell_config_replace(m_config, "encoding", "utf-8");

-#ifdef Q_OS_WIN
-    QString aspell_data_dir();
-    aspell_config_replace(m_config, "data-dir", aspell_data_dir().toLocal8Bit().data());
-    aspell_config_replace(m_config, "dict-dir", aspell_data_dir().toLocal8Bit().data());
-#endif
-
     AspellCanHaveError *possible_err = new_aspell_speller(m_config);

     if (aspell_error_number(possible_err) != 0) {
--
2.15.0.windows.1

